### PR TITLE
Allow to also validate street, company type, etc

### PR DESCRIPTION
**Pull request by Freeaqingme**
My customer also wanted to check against street and company type, etc (for countries that support this). Figured we could just as well upstream it...

### CODE REVIEW

**Comment by DragonBe** [comment]
Hey @Freeaqingme, thank you for your PR. It was something I was thinking about, but never got around to add. 👍 

I've looked at your proposed changes but I need to analyse it a little more in-depth to ensure the provided logic is not in danger. I will get back to you in a few days when I've validated your changes.

File: src/Vies/CheckVatResponse.php [hunk 1]
```diff
@@ -262,6 +468,16 @@ public function populate($row): void
...
+
+            ->setStreet($row->traderStreet ?? '---')
+            ->setPostcode($row->traderPostcode ?? '---')
+            ->setCity($row->city ?? '---')
+            ->setCompanyType($row->companyType ?? '---')
+            ->setNameMatch($row->traderNameMatch ?? self::NOT_PROCESSED)
+            ->setCompanyTypeMatch($row->traderCompanyTypeMatch ?? self::NOT_PROCESSED)
+            ->setStreetMatch($row->traderStreetMatch ?? self::NOT_PROCESSED)
+            ->setPostcodeMatch($row->traderPostcodeMatch ?? self::NOT_PROCESSED)
+            ->setCityMatch($row->traderCityMatch ?? self::NOT_PROCESSED)
```
Comment (DragonBe) [inline src/Vies/CheckVatResponse.php]: This isn't going to work as it requires population of the fields through the row returned. Best is to have the default values set at construct and override with data if it's available.

Comment (DragonBe) [reply src/Vies/CheckVatResponse.php]: To give you an idea:

```lang=bash
1) DragonBe\Test\Vies\CheckVatResponseTest::testResponseContainsEmptyValuesWithOnlyRequiredArguments with data set #0 ('DE', '123456749', '2018-09-20+02:00', true)
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
     'name' => '---'
     'address' => '---'
     'identifier' => ''
+    'street' => '---'
+    'postcode' => '---'
+    'city' => '---'
+    'companyType' => '---'
+    'nameMatch' => 3
+    'companyTypeMatch' => 3
+    'streetMatch' => 3
+    'postcodeMatch' => 3
+    'cityMatch' => 3
 )

/Users/dragonbe/workspace/dragonbe/vies/tests/Vies/CheckVatResponseTest.php:231

2) DragonBe\Test\Vies\CheckVatResponseTest::testResponseContainsEmptyValuesWithOnlyRequiredArguments with data set #1 ('ES', '987654321', '2018-09-20+02:00', false)
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
     'name' => '---'
     'address' => '---'
     'identifier' => ''
+    'street' => '---'
+    'postcode' => '---'
+    'city' => '---'
+    'companyType' => '---'
+    'nameMatch' => 3
+    'companyTypeMatch' => 3
+    'streetMatch' => 3
+    'postcodeMatch' => 3
+    'cityMatch' => 3
 )

/Users/dragonbe/workspace/dragonbe/vies/tests/Vies/CheckVatResponseTest.php:231
```

**Comment by Freeaqingme** [comment]
Thanks @DragonBe . Though I hadn't forgotten about it yet, I also clearly hadn't prioritized it. Besides, your implementation looks at a glance to be more elaborate than mine. Way to go!

**Comment by DragonBe** [comment]
You’re welcome @Freeaqingme. Just double check if the return values for trader validation work out for you as they differ from the specs.

**Pull request closed without merging**